packages/cli/src/rpc.ts

```diff
@@ -1,17 +1,38 @@
 import debug from 'debug'
+import { PortalNetwork, getContentId } from 'portalnetwork'
+import { fromHex } from '@chainsafe/discv5'
 
 const log = debug('RPC')
 
-export const methods = {
-  discv5_nodeInfo: async () => {
-    log('discv5_nodeInfo request received')
-    return 'Ultralight-CLI: v0.0.1'
-  },
-  eth_getBlockByHash: async (params: [string, boolean]) => {
-    const [blockHash, includeTransactions] = params
-    log(
-      `eth_getBlockByHash request received. blockHash: ${blockHash} includeTransactions: ${includeTransactions}`
-    )
-    return 'Ultralight-CLI: v0.0.1'
-  },
+export class RPCManager {
+  public _client: PortalNetwork
+  private _methods: { [key: string]: Function } = {
+    discv5_nodeInfo: async () => {
+      log('discv5_nodeInfo request received')
+      return 'Ultralight-CLI: v0.0.1'
+    },
+    eth_getBlockByHash: async (params: [string, boolean]) => {
+      const [blockHash, includeTransactions] = params
+      log(
+        `eth_getBlockByHash request received. blockHash: ${blockHash} includeTransactions: ${includeTransactions}`
+      )
+      const lookupKey = getContentId({ chainId: 1, blockHash: fromHex(blockHash.slice(2)) }, 0)
+      const value = await this._client.db.get(lookupKey)
+      if (value) {
+        return value
+      }
+      // TODO: Add sendFindContent to this to retrieve from network if not found in local DB
```

Comment:
Lets maybe finish this `sendFindContent` piece for now where we retrieve the block header from the network and then return it in the correct format (either RLP or JSON blob, whichever is according to the JSON-RPC spec).  We can add the block bodies part later.

Reply:
Sounds good, will keep working on it. Just checked the specs and the content response should be RLP - https://github.com/ethereum/portal-network-specs/blob/master/history-network.md?plain=1#L61

Reply:
For sure in the sendFindContent response.  This is actually what I was talking about - https://eth.wiki/json-rpc/API#eth_getBlockByHash  We need to mirror the standard Ethereum JSON RPC responses for methods with the same names.  Think of this RPC as a drop-in replacement for a full node in the set of method we are aiming to support.